### ISSUE TITLE
Specify additional root paths

### ISSUE THREAD

**Issue by Yogu** [ng-packagr/ng-packagr#1077]
## Type of Issue

```
[ ] Bug Report
[x] Feature Request
```

## Description

I'd like to extend TypeScript's root paths with a `.d.ts` file that adds module declarations. Within angular apps (with the webpack setup), this is possible by adding it to `files` in `tsconfig`. ng-packagr overrides the `rootPaths` property, thus ignoring both `include` and `files` (correct me if I'm wrong).

It would be nice if the `files` specified in the `tsconfig` file would just be extended with the entry file path (e.g. `public_api.ts`) instead of being completely overridden.

I need this because I import from untyped modules that do not have a DefinitelyTyped package. The recommended solution to this is to just include a `.d.ts` file in the compilation that has a `declare module` declaration. This is only possible in `.d.ts` files, so I can't just put it in a TypeScript file that is imported from the entry file. It is possible to reference `.d.ts` files via `/// <reference>`, but this is not compatible with some third-party tools. TypeScript also allows to specify module aliases via `path` but this just opens another can of worms and causes issues with the angular webpack plugin.

In summary, the most straight-forward way to include custom module typings is to add them to `include` / `files`.

#### How To Reproduce

Specify `files` within a `tsconfig` file used with ng-packagr.

#### Expected Behaviour

The specified files should be included in the compilation context.

#### Version Information

```
$ node_modules/.bin/ng-packagr --version
ng-packagr:            3.0.6
@angular/compiler:     6.1.0
rollup:                0.59.4
tsickle:               0.32.1
typescript:            2.9.2
```

_Please include all version numbers that might be relevant, e.g. third-party libraries_

**Comment by Yogu** [comment]
Triple slash imports usually work, but there are a few problems using them for package types:

- You can't place the target files outside of the library's root, otherwise, they will not be copied to the dist directory and the reference will point to nothing. Thus, in a multi-library repo, you need to copy the type declarations to all libraries that use them.
- The package types will be shipped with your library, which is probably not what you want.
    - you may not even need the typings in the API surface of your library
    - users of your library are forced to use your typings and can't easily override them. This is bad if you just declared the types in a hacky way for the parts needed in your library.
    - if multiple packages ship the types, users will get a "duplicate identifier" error.

In summary, I need a way to add declaration files to the compilation context without emitting them or a reference to them. In a regular TypeScript project, this works by adding it to `include` / `files`.
